library/Rules/CreditCardVisa.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of Respect/Validation.
+ *
+ * (c) Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ *
+ * For the full copyright and license information, please view the "LICENSE.md"
+ * file that was distributed with this source code.
+ */
+
+namespace Respect\Validation\Rules;
+
+class CreditCardVisa extends CreditCard
+{
+
+    public function validate($input)
+    {
+        $input = preg_replace('([^0-9])', '', $input);
```

Comment:
I was thinking about an abstract class with an abstract method where this method would have the code to validate a certain brand, so you would inherit from this abstract class and just code the part about the brand itself. doing so, the validate method would be untouchable. 

what do you think?